File: src/main/java/de/pflugradts/pwman3/domain/model/password/Key.java
package de.pflugradts.pwman3.domain.model.password;

import de.pflugradts.pwman3.domain.model.ddd.DomainEntity;
import de.pflugradts.pwman3.domain.model.transfer.Bytes;
import lombok.AccessLevel;
import lombok.EqualsAndHashCode;
import lombok.Getter;

/**
 * A Key identifies a {@link Password}.
 */
@EqualsAndHashCode
final class Key implements DomainEntity {

    @Getter(AccessLevel.PRIVATE)
    private Bytes bytes;

    private Key(final Bytes bytes) {
        this.bytes = bytes.copy();
    }

    static Key create(final Bytes bytes) {
        return new Key(bytes);
    }

    void rename(final Bytes newBytes) {
        this.bytes = newBytes.copy();
    }

    Bytes view() {
        return getBytes().copy();
    }

}

File: src/main/kotlin/de/pflugradts/passbird/domain/model/password/Key.kt
package de.pflugradts.passbird.domain.model.password

import de.pflugradts.passbird.domain.model.ddd.DomainEntity
import de.pflugradts.passbird.domain.model.transfer.Bytes

/**
 * A Key identifies a [Password].
 */
class Key private constructor(private var bytes: Bytes) : DomainEntity {
    fun rename(newBytes: Bytes) { bytes = newBytes.copy() }
    fun view() = bytes.copy()
    override fun equals(other: Any?) = (other as? Key)?.view() == bytes
    override fun hashCode() = bytes.hashCode()
    companion object {
        fun createKey(bytes: Bytes) = Key(bytes.copy())
    }
}
